### PR TITLE
Moving field named INDEX from LayoutFileSystem to MavenRepositoryFeat…

### LINKED ISSUE #1590: Drop the INDEX field from the LayoutFileSystem
# Task Description

We need to drop the `INDEX` field from the `LayoutFileSystem`, as this is specific only for the Maven layout provider. Perhaps a better place for this would be the `MavenRepositoryFeatures` class.

# Tasks

The following tasks will need to be carried out:
* [ ] Refactor the `LayoutFileSystem`, so that it no longer contains the `INDEX`.
* [ ] Make sure none of the other (non-Maven) layout providers are affected.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by anki2189**
…ures

# Pull Request Description

This pull request closes #1590

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No
#1590

### CODE REVIEW

**Comment by anki2189** [comment]
@carlspring Pls review the changes

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-import javax.inject.Inject;
-import java.io.IOException;
-import java.time.Instant;
-import java.time.temporal.ChronoUnit;
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
-
-import org.apache.commons.lang3.BooleanUtils;
-import org.apache.maven.index.artifact.M2ArtifactRecognizer;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java]: Cosmetic: Could you please set up your IDE to not reformat imports unless necessary and to conform with our [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html)? Thanks! :)

Comment (anki2189) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java]: I have imported [code style tempalte](https://strongbox.github.io/developer-guide/coding-convention.html) will set up IDE to not reformat imports and raise PR again.

**Comment by carlspring** [comment]
No need for a new PR. Just push to this one.

**Comment by carlspring** [comment]
Thanks! :)

**Pull request merged**